### DISCUSSION TITLE
Single sided BOB for STM32F4xx Black pill

### DISCUSSION THREAD

**Discussion by avizienis** [terjeio/grblHAL#219] [General]
Hi everyone, just finished testing my ghetto version of BOB for the popular Black pill by WeAct. Was wondering if anyone would be interested in the design files.
Properties:
1. 4 axes outputs (3 limit inputs);
2. Single sided PCB;
3. Optocoupled inputs;
4. Buffered 5V outputs;
5. Onboard downconverter. Could be 7805, I used a micro DC-DC converter.
It's based on earlier version of Phil's BOB but stripped down to the bare minimum. To minimize routing I had to adjust the existing map files slightly, so that all outputs (except PWM) are on one side of the chip, all inputs on the other.
![1611848400086s](https://user-images.githubusercontent.com/7872840/106163005-5e966980-6191-11eb-83ee-cb61e07f09fd.jpg)
![1611848400070s](https://user-images.githubusercontent.com/7872840/106162965-550d0180-6191-11eb-8173-3ac8c2bf7ffd.jpg)

**Comment by terjeio** [comment]
That looks good!
If you publish the design I will link to it on the driver page. If you are going to do so you should give the board a name and make a PR against the test branch for a new map file for the board.

**Reply by avizienis** [reply]
Hi Terje, all project files have been uploaded to:
https://github.com/avizienis/Minimal-Black-Pill--STM32F4xx-BOB-for-grblHAL
I will also try to create a PR in a sec

**Reply by terjeio** [reply]
I see you have added a map file to your repo. Can I add it to [the driver](https://github.com/grblHAL/STM32F4xx/tree/master/Inc) and add entries in _my_machine.h_ and _driver.h_ for it?

**Reply by avizienis** [reply]
Yes absolutely, I remember creating a pull request for exactly that some time ago

**Reply by terjeio** [reply]
> Yes absolutely, I remember creating a pull request for exactly that some time ago

You made a direct commit to the repo, not a PR - I have missed that. It was lost when I moved the code to the [new home](https://github.com/grblHAL) from my subversion master - I'll fix that.
